pypesto/sample/dynesty.py

```diff
@@ -149,6 +163,27 @@ def initialize(
         if self.dynamic:
             sampler_class = dynesty.DynamicNestedSampler
 
+        # check if objective fits to the pyPESTO problem
+        if self.objective_type == "neglogpost":
+            if self.problem.x_priors is None:
+                # objective is the negative log-posterior, but no priors are defined
+                # sampler needs the likelihood
+                raise ValueError(
+                    "x_priors have to be defined in the problem if objective is 'neglogpost'."
+                )
+        else:
+            # if objective is the negative log likelihood, we will ignore x_priors even if they are defined
+            if self.problem.x_priors is not None:
+                logger.info(
+                    "Warning: Assuming 'negloglike' as objective. 'x_priors' defined in the problem will be ignored."
+                )
+
+        # if priors are uniform, we can use the default prior transform (assuming that bounds are set correctly)
+        logger.info(
+            "Assuming 'prior_transform' is correctly specified. If 'x_priors' is not uniform, 'prior_transform'"
+            " has to be adjusted accordingly."
+        )
```

Comment:
I'll try to make this nicer, we should be able to support e.g. Gaussian transform at least

Reply:
defining the `prior_transfromation`, if mean and cov of the Gaussian prior are known, would be easy. However, I am not so sure how to extract these information out of the current pyesto problem/objective.

Reply:
OK, then let's wait until pyPESTO has better objective attributes